ministryofjustice/analytics-platform-ops
# Adds fluentd container to shiny-app Helm chart, to ship logs to Elasticsearch

## What

Adds a fluentd container to Shiny apps, to ship application logs to elasticsearch cluster

## How to review

1. Check out this branch
2. Deploy a shiny app using the shiny-app Helm chart:
`helm upgrade -i rshiny-example-1 charts/shiny-app --namespace apps-prod --set app.repo=rshiny-example --set app.name=rshiny-example-1 --set app.baseHost=apps.ENV.BASEDOMAIN --set shinyApp.docker.repository=ECR-REPOSITORY/rshiny-example --set shinyApp.docker.tag=IMAGE_TAG --set OAuth.ClientSecret=CLIENT_SECRET --set OAuth.ClientId=CLIENT_ID --set OAuth.Domain=AUTH0_DOMAIN --set CookieSecret=XXXChangeThisXXX --set Elasticsearch.Host=ELASTICSEARCH_DOMAIN --set Elasticsearch.Port=ELASTICSEARCH_PORT`
3. Run Jenkins grant access job to add a user
4. Access app in browser
5. Access kibana and observe application logs

**NOTE**: as of now the Jenkins grant access job does not work, so a user would have to be added in Auth0 manually
